lib/runner/TestRunner.js

```diff
@@ -0,0 +1,171 @@
+        const testResults = [];
+        for (const test of testSuite.tests) {
+            invoker.beforeTest(test);
+
+            const testResult = new TestResult(test);
+            testResults.push(testResult);
+            if (test.skip) {
+                continue;
+            }
+
+            // Process through the interactions
+            // Look for goto statements and exits, which will affect the sequence
+            let goto;
+            for (const interaction of test.interactions) {
+                // If a goto is set, keep skipping until we match it
+                if (goto) {
+                    if (goto === interaction.utterance) {
+                        goto = undefined;
+                    } else {
+                        continue;
+                    }
+                }
+
+                const interactionResult = await this.interact(invoker, testSuite, test, interaction);
+                if (interactionResult.goto) {
+                    // If this result is a goto, set the goto label
+                    goto = interactionResult.goto;
+                }
+
+                testResult.addInteractionResult(interactionResult);
+
+                if (interactionResult.exited) {
+                    // If this is an exit, stop processing
+                    break;
+                }
+            }
+
+            invoker.afterTest(test);
+        }
+
+        invoker.after(testSuite);
+        return testResults;
+    }
+
+    async interact(invoker, testSuite, test, interaction) {
+        if (testSuite.trace) {
+            // eslint-disable-next-line no-console
+            console.log("File: " +  testSuite.shortFileName + " Test: " + test.description + " Utterance: " + interaction.utterance);
+        }
+
+        let response;
+        try {
+            response = await invoker.invoke(interaction);
+        } catch (e) {
+            // We do special-handling for these errors
+            if (e.message.startsWith("Unable to match utterance:")
+                || e.message.startsWith("Interaction model has no intentName named")) {
+                const message = Util.errorMessageWithLine(e.message, testSuite.fileName, interaction.lineNumber);
+                return new InteractionResult(interaction, undefined, message);
+            } else {
+                return new InteractionResult(interaction, undefined, e.message + "\n" + e.stack);
+            }
+        }
+
+        // Add shorthand values like prompt and re-prompt to the response
+        response.inject();
+
+        let result = new InteractionResult(interaction);
+        for (const assertion of interaction.assertions) {
+            if (assertion.exit) {
+                result = new InteractionResult(interaction, assertion);
+                break;
+            }
+
+            if (!response.supported(assertion.path)) {
+                continue;
+            }
+
+            const passed = assertion.evaluate(response);
```

Comment:
I was going to comment to change response.json to response, but you already did it. Guess I better work on top of your PR